submissions/Kostiantyn_Rakhmanov/frogger/app.js

```diff
@@ -0,0 +1,138 @@
+const CANVAS = {
+    start: 0,
+    rows: 6,
+    columns: 5,
+    section: {
+        width: 100,
+        height: 80,
+    },
+};
+
+const CANVAS_WIDTH = CANVAS.section.width * CANVAS.columns;
+const CANVAS_HEIGHT = CANVAS.section.height * CANVAS.rows;
+
+const PLAYER = {
+    startCoordX: 200,
+    startCoordY: 400,
+    startLocationShift: 60,
+    startDelay: 777,
+};
+
+const ENEMY = {
+    coordX: {
+        min: -100,
+        max: 510,
+    },
+    speed: {
+        min: 50,
+        max: 100,
+    },
+    collision: {
+        x: 70,
+        y: 60,
+    },
+    locationsY: [
+        60,
+        145,
+        225,
+    ],
+};
+
+var Enemy = function (x, y, speed, player) {
+    this.x = x;
+    this.y = y;
+    this.sprite = 'images/enemy-bug.png';
+    this.speed = speed;
+    this.player = player;
+};
+
+Enemy.prototype.update = function (dt) {
+    this.x += this.speed * dt;
+    if (this.x > ENEMY.coordX.max) {
+        this.x = ENEMY.coordX.min;
+        this.speed = ENEMY.speed.min + Math.floor(Math.random() * ENEMY.speed.max);
+    }
+    this.checkCollision();
+};
+
+Enemy.prototype.checkCollision = function () {
+    if (this.player.x < this.x + ENEMY.collision.x &&
+        this.player.x + ENEMY.collision.x > this.x &&
+        player.y < this.y + ENEMY.collision.y &&
+        ENEMY.collision.y + player.y > this.y) {
+        this.player.toStart();
+    }
+};
+
+Enemy.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+let Player = function (x, y, enemies) {
+    this.x = x;
+    this.y = y;
+    this.sprite = 'images/char-boy.png';
+    this.enemies = enemies;
+    this.startX = 200;
+    this.startY = 400;
```

Comment:
Any correlation with CANVAS properties?
So if canvas dimensions change as a peer developer I will need to change numbers in multiple locations instead of relying on programmed automation and calculations.
DRY and constants help to avoid this problem.
Reflect on why we make code DRY and why we use contstants. How these two concepts intersect.